src/protocol.c

```diff
@@ -62,13 +63,51 @@ bool check_id(const char *id) {
 	return true;
 }
 
+static bool check_request_type( struct meshlink_handle *mesh, const char *request, submesh_t **from, submesh_t **to ) {
```

Comment:
We shouldn't need to parse every request we send via `send_request()` or `forward_request()`, that will be bad for performance. Also, why do we only limit broadcasts of ADD_EDGE messages? We should filter *all* messages to ensure nodes in a sub-mesh only see messages from their own sub-mesh or from the parent mesh. The exception is when we call `send_add_edge(..., mesh->everyone, ...)` and `send_del_edge(..., mesh->everyone, ...)`, and when we call `send_everything()`. In case we send add/del messages to everyone, we should only send requests to the proper nodes. Perhaps `send_request()` should get an extra parameter names `submesh`, and if `c == mesh->everyone`, it should call `broadcast_submesh_meta()` with that `submesh`.

Reply:
I have changed the send_request api and also forward_request api according to your suggestion and removed the request processing part from send_request api